kafka-rest/src/main/java/io/confluent/kafkarest/entities/ProduceResult.java

```diff
@@ -28,11 +31,32 @@
 
   public abstract long getOffset();
 
-  public static ProduceResult create(int partitionId, long offset) {
-    return new AutoValue_ProduceResult(partitionId, offset);
+  public abstract Optional<Instant> getTimestamp();
+
+  public abstract int getSerializedKeySize();
+
+  public abstract int getSerializedValueSize();
+
+  public static ProduceResult create(
+      int partitionId,
+      long offset,
+      @Nullable Instant timestamp,
```

Comment:
why nullable?

Reply:
As pointed out somewhere else, timestamp is optionally returned by Kafka.